modules/stochastic_tools/include/userobjects/InverseMapping.h

```diff
@@ -0,0 +1,73 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+// MOOSE includes
+#include "GeneralUserObject.h"
+#include "MappingInterface.h"
+#include "SurrogateModelInterface.h"
+#include "SurrogateModel.h"
+
+/**
+ * A user object which takes a surrogate (or just user supplied values)
+ * to determine coordinates in a latent space and uses those coordinates
+ * to create approximations of full solution values for given variables.
+ */
+class InverseMapping : public GeneralUserObject,
```

Comment:
Yes and met strong resistance. AuxKernel goes on an element/nodal loop and I would like to do this in one go using vectors and dofs. This would be general enough to handle Monomial,FV,Lagrange bases. I would probably have to create two different AuxKernels for Monomial/FV and Lagrange.

Reply:
Since Daniel's PR we can make auxkernels fill multiple auxvariables.

However I m not sure I like this, and Alex has also expressed his opposition

We could have one UO that does the work and several auxkernels to do the final steps, or just this UO

Reply:
Correct, I don't like it 😄 I like it fine from UOs, but hate it from aux kernels